File: src/TheManeFeed.Web/projects/api/src/lib/models/article.ts
export interface ArticleListItem {
  id: number;
  title: string;
  summary: string | null;
  imageUrl: string | null;
  sourceName: string;
  url: string;
  category: string | null;
  categorySlug: string | null;
  author: string | null;
  publishedAt: string | null;
  readCount: number;
  isFeatured: boolean;
  isTrending: boolean;
}

export interface ArticleAuthor {
  id: number;
  name: string;
  avatarUrl: string | null;
  bio: string | null;
}

export interface ArticleDetail {
  id: number;
  title: string;
  summary: string | null;
  body: string | null;
  imageUrl: string | null;
  sourceName: string;
  url: string;
  category: string | null;
  categorySlug: string | null;
  author: ArticleAuthor | null;
  publishedAt: string | null;
  scrapedAt: string;
  readCount: number;
  isFeatured: boolean;
  isTrending: boolean;
}

export interface CategoryArticleItem {
  id: number;
  title: string;
  summary: string | null;
  imageUrl: string | null;
  sourceName: string;
  url: string;
  author: string | null;
  publishedAt: string | null;
  readCount: number;
}

File: src/TheManeFeed.Core/Entities/Article.cs
namespace TheManeFeed.Core.Entities;

public class Article
{
    public int Id { get; set; }
    public string Url { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Summary { get; set; }
    public string? Body { get; set; }
    public string? ImageUrl { get; set; }
    public string SourceName { get; set; } = string.Empty;
    public int? CategoryId { get; set; }
    public int? AuthorId { get; set; }
    public DateTime? PublishedAt { get; set; }
    public DateTime ScrapedAt { get; set; } = DateTime.UtcNow;
    public int ReadCount { get; set; }
    public bool IsFeatured { get; set; }
    public bool IsTrending { get; set; }

    public Category? Category { get; set; }
    public Author? Author { get; set; }
    public ICollection<SavedArticle> SavedByUsers { get; set; } = new List<SavedArticle>();
    public ICollection<CollectionArticle> CollectionArticles { get; set; } = new List<CollectionArticle>();
}
